sqlx-core/src/postgres/type_info.rs

```diff
@@ -995,20 +995,18 @@ impl PartialEq<PgType> for PgType {
         if let (Some(a), Some(b)) = (self.try_oid(), other.try_oid()) {
             // If there are OIDs available, use OIDs to perform a direct match
             a == b
+        } else if (matches!(self, PgType::DeclareWithName(_))
+            && matches!(other, PgType::DeclareWithOid(_)))
+            || (matches!(other, PgType::DeclareWithName(_))
+                && matches!(self, PgType::DeclareWithOid(_)))
```

Comment:
This can be further simplified I think as: 
```rust
if let 
    (PgType::DeclareWithName(_), PgType::DeclareWithOid(_)) 
        | (PgType::DeclareWithOid(_),PgType::DeclareWithName(_)) = (self, other) 
{
```

(or however `rustfmt` likes it)